goal_src/engine/camera/cam-layout.gc

```diff
@@ -5,3 +5,5155 @@
+         (set! (-> a0-7 x) 192)
+         (set! (-> a0-7 y) 0)
+         (set! (-> a0-7 z) 0)
+         0
+         )
+        )
+       (camera-line-setup (the-as vector a0-7))
+       )
+      (let ((s4-0 (the-as object (-> s5-0 first-point))))
+       (dotimes (s3-0 (/ (-> s5-0 point-count) 2))
+        (camera-line-draw (the-as vector s4-0) (&+ (the-as vector s4-0) 16))
+        (set!
+         s4-0
+         (&-> (the-as (inline-array plane-volume) s4-0) 1 first-point)
+         )
+        )
+       )
+      )
+     )
+    )
+   )
+  #f
+  )
+
+(defun v-slrp! ((arg0 vector) (arg1 vector) (arg2 vector) (arg3 float))
+  (let ((s2-0 (new-stack-vector0))
+        (s1-0 (new-stack-vector0))
+        (s0-0 (new-stack-vector0))
+        )
+   0.0
+   0.0
+   0.0
+   0.0
+   (cond
+    ((< (-> (new 'static 'array float 1 1.0) 0) arg3)
+     (set!
+      arg3
+      (-> (the-as (pointer float) (new 'static 'array float 1 1.0)) 0)
+      )
+     )
+    ((< arg3 0.0)
+     (set!
+      arg3
+      (-> (the-as (pointer float) (new 'static 'array float 1 0.0)) 0)
+      )
+     )
+    )
+   (vector-normalize-copy! s2-0 arg1 (-> (new 'static 'array float 1 1.0) 0))
+   (vector-normalize-copy! s1-0 arg2 (-> (new 'static 'array float 1 1.0) 0))
+   (vector-cross! s0-0 s2-0 s1-0)
+   (let* ((f30-0 (vector-length s0-0))
+          (f28-0 (asin f30-0))
+          )
+    (vector-float*!
+     arg0
+     arg1
+     (/ (sin (* (- (-> (new 'static 'array float 1 1.0) 0) arg3) f28-0)) f30-0)
+     )
+    (vector+float*! arg0 arg0 arg2 (/ (sin (* arg3 f28-0)) f30-0))
+    )
+   )
+  )
+
+(deftype interp-test-info (structure)
+  ((from   vector   :inline :offset-assert   0)
+   (to     vector   :inline :offset-assert  16)
+   (origin vector   :inline :offset-assert  32)
+   (color  vector4w         :offset-assert  48)
+   (axis   vector           :offset-assert  52)
+   (disp   symbol           :offset-assert  56)
+   )
+  :method-count-assert 9
+  :size-assert         #x3c
+  :flag-assert         #x90000003c
+  )
+
+
+(defun
+  interp-test
+  ((arg0 (function vector vector vector float vector none))
+   (arg1 interp-test-info)
+   )
+  (let ((s3-0 (new-stack-vector0))
+        (gp-0 (new-stack-vector0))
+        )
+   (let ((t9-0 arg0)
+         (a0-1 s3-0)
+         (a1-1 (-> arg1 from))
+         (a2-0 (-> arg1 to))
+         (a3-0 (-> (new 'static 'array float 1 0.0) 0))
+         (t0-0 (-> arg1 axis))
+         )
+    (-> (new 'static 'array float 1 65536.0) 0)
+    (t9-0 a0-1 a1-1 a2-0 a3-0 t0-0)
+    )
+   (vector+! s3-0 s3-0 (-> arg1 origin))
+   (dotimes (s2-0 10)
+    (set! (-> gp-0 quad) (-> s3-0 quad))
+    (let ((t9-1 arg0)
+          (a0-5 s3-0)
+          (a1-4 (-> arg1 from))
+          (a2-1 (-> arg1 to))
+          (a3-1
+           (* 0.1 (+ (-> (new 'static 'array float 1 1.0) 0) (the float s2-0)))
+           )
+          (t0-1 (-> arg1 axis))
+          )
+     (-> (new 'static 'array float 1 65536.0) 0)
```

Comment:
I'm guessing the `arg0` function takes one more additional `float` argument here